### PR TITLE
Added support for FetchRequest protocol version 3.

### PR DESCRIPTION
**Pull request by sp1ff**
This commit will add the 'MaxBytes' field to FetchRequests when the Kafka version is
0.10.1 or better (i.e. it adds support for V3 of the FetchRequest API). On send, it will be set to MaxResponseSize.

### CODE REVIEW

File: fetch_request.go [hunk 1]
```diff
@@ -32,6 +36,9 @@ func (r *FetchRequest) encode(pe packetEncoder) (err error) {
 	pe.putInt32(-1) // replica ID is always -1 for clients
 	pe.putInt32(r.MaxWaitTime)
 	pe.putInt32(r.MinBytes)
+	if 3 == r.Version {
```
Comment (eapache) [inline fetch_request.go]: `r.Version == 3` is more consistent with the rest of the code, and with the check you added below

File: consumer.go [hunk 2]
```diff
@@ -726,6 +726,10 @@ func (bc *brokerConsumer) fetchNewMessages() (*FetchResponse, error) {
 	if bc.consumer.conf.Version.IsAtLeast(V0_10_0_0) {
 		request.Version = 2
 	}
+	if bc.consumer.conf.Version.IsAtLeast(V0_10_1_0) {
+		request.Version = 3
+		request.MaxBytes = MaxResponseSize - 47 // TODO(mherstin): WTF?!
```
Comment (eapache) [inline consumer.go]: I agree with the comment: where did the value of 47 come from?

File: fetch_request.go (hunk 1 above)
Comment (sp1ff) [inline fetch_request.go]: Old habits die hard; fixed in my latest commit.

File: consumer.go (hunk 2 above)
Comment (sp1ff) [inline consumer.go]: That's embarrassing-- this was removed in a subsequent commit which I forgot to push. Fixed.

**Comment by eapache** [comment]
Thanks!

**Pull request merged**